File: uptime3.cpp
#include <stdio.h>
#include "strd.hpp"
int main(){
	//getting time from the uptime file in unix
	readf uptimefile("/proc/uptime");
	int time = std::stoi(uptimefile.read());
	
	int weeks   = time    / (7*24*60*60);
	int days    = time    / (24*60*60)   - 7*weeks;
	int hours   = time    / (60*60)      - 7*24*weeks    -24*days;
	int minutes = time    / (60)         - 7*24*60*weeks -24*60*days - 60*hours; 	 

	std::string uptime = "Up Time: ";
	
	if(weeks !=0)
		if(weeks==1)
			uptime += std::to_string(weeks) + " week, ";
		else
			uptime += std::to_string(weeks) + " weeks, ";
	
	if(days !=0)
		if(days==1)
			uptime += std::to_string(days)  + " day, ";
		else
			uptime += std::to_string(days)  + " days, ";
				
	if(hours !=0)
		if(hours==1)
			uptime += std::to_string(hours) + " hour, ";
		else
			uptime += std::to_string(hours) + " hours, ";

	if(minutes !=0)
		if(minutes==1)
			uptime += std::to_string(minutes) + " minute.";
		else
			uptime += std::to_string(minutes) + " minutes.";

	
	std::cout<<uptime<<std::endl;
}

File: uptime3.py
#!/usr/bin/python3
# Nick Chirico
# Oct 6, 2016

import datetime
import calendar
import sys

# Given the uptime in seconds, prints the date and time the system came up.
def origin(time):
	delta = datetime.timedelta(seconds=time)
	origin = str ( datetime.datetime.now() - delta )
	formatted = origin.replace(" ","-")
	formatted = formatted.replace(":","-")
	formatted = formatted.replace(":","-")
	formatted = formatted.split(".",1)[0]
	formatted = formatted.split("-")

	# Get month from provided number
	since = calendar.month_name[int(formatted[1])]
	# Everyhting date related
	since += " " + formatted[2].replace("0","") + " " + formatted[0] + " at "
	# everything time related
	since += formatted[3] + ":" + formatted[4] + "."

	print("Up since:", since)

def up(time):
	# Actual Calculations
	## Minutes
	minutes = int (( time/60 ) % 60 )
	overflow = int (( time/60 ) - ( minutes ))
	## Hours
	hours = int (( overflow/60 ) % 24 )
	overflow = int (( overflow/60 ) - ( hours ))
	## Days
	days = int (( overflow/24 ) % 24 )
	overflow = int (( overflow/24 ) - ( days ))
	## Weeks
	weeks = int (( overflow/7 ) % 7 )


	# This will be printed later. Concat on relevant info
	output = "Uptime: "
	if weeks >= 1:
		output = output + str ( int ( weeks ) ) + " weeks, "
	if days >= 1:
		output = output + str ( int ( days ) ) + " days, "
	if hours >= 1:
		output = output + str ( int ( hours ) ) + " hours, "
	if minutes >= 1:
		output = output + str ( int ( minutes ) ) + " minutes."
	# Let's be proper now. In case this is done on the hour mark.
	else:
		output += "."
	print(output)
